llama-index-integrations/readers/llama-index-readers-microsoft-sharepoint/llama_index/readers/microsoft_sharepoint/base.py

```diff
@@ -7,37 +7,47 @@
 
 import requests
 from llama_index.core.readers import SimpleDirectoryReader
-from llama_index.core.readers.base import BaseReader
+from llama_index.core.readers.base import BasePydanticReader
 from llama_index.core.schema import Document
 
 logger = logging.getLogger(__name__)
 
 
-class SharePointReader(BaseReader):
+class SharePointReader(BasePydanticReader):
     """SharePoint reader.
 
+
     Reads folders from the SharePoint site from a folder under documents.
+
+    Args:
+        client_id (str): The Application ID for the app registered in Microsoft Azure Portal.
+            The application must also be configured with MS Graph permissions "Files.ReadAll", "Sites.ReadAll" and BrowserSiteLists.Read.All.
+        client_secret (str): The application secret for the app registered in Azure.
+        tenant_id (str): Unique identifier of the Azure Active Directory Instance.
     """
 
+    client_id: str = None
+    client_secret: str = None
+    tenant_id: str = None
+    _authorization_headers = None
```

Comment:
if this is a Pydantic `BaseModel`, should this be `PrivateAttr`?

Reply:
yup good catch